cli/cli.go

```diff
@@ -508,13 +508,21 @@ func (runOptions *runOptionsType) handleCLIOptions(ctx *cli.Context) error {
 	switch ctx.Command.Name {
 	case "install",
 		"bootstrap",
-		"show-artifact",
-		"show-provides":
+		"show-artifact":
 		log.Debug("calling doHandleBootstrapArtifact")
 		err = doHandleBootstrapArtifact(config, runOptions)
 		if err != nil {
 			log.Errorf("Error while handling bootstrap Artifact, continuing: %s", err.Error())
 		}
+	case "show-provides":
+		_, err = os.Stat(config.BootstrapArtifactFile)
+		if err == nil {
+			log.Debug("calling doHandleBootstrapArtifact")
+			err = doHandleBootstrapArtifact(config, runOptions)
+			if err != nil {
+				log.Errorf("Error while handling bootstrap Artifact, continuing: %s", err.Error())
+			}
+		}
```

Comment:
This won't work correctly, you are skipping the entire handling, including the handling for when the bootstrap artifact is not present. With this the client would not get any name if the artifact is not present, and would probably misbehave on the first update.

The problem is likely somewhere in [this function](https://github.com/mendersoftware/mender/blob/master/app/mender.go#L667), but it still has to do its job.

Reply:
okey, I need some education on the bootstrap artifact.

Reply:
my reasoning was: if the bootstrap artifact file does not exist then there is no point in calling `doHandleBootstrapArtifact`, but apparently I missed something?

Reply:
Yes, that case is handled inside the function I mentioned. It then needs to commit the `unknown` name, and prevent the bootstrap artifact from being considered later by writing an entry into the database.

My gut feeling is that the speed has something to do with the database loading, and especially `LoadStateData`, which requires a write transaction, even when reading (long story). This could be optimized, but it would be useful to know whether this actually is the problem.

Perhaps you could time a few key points in that function? The most likely candidates are `LoadStateData` and `LoadProvides`, but there could of course be something else as well.

Reply:
ok, but if the file does not exist, `HandleBootstrapArtifact` returns nil [at mender.go:749](https://github.com/mendersoftware/mender/blob/master/app/mender.go#L749) and does nothing.

Reply:
okey, discussed privately. I am on it in other words.

Reply:
okey, @kacf what do you think now? I want to add tests, make it prettier, move some functions a bit, but the general idea is here.

Reply:
correction: I messed up: @kacf I will ping you for a review, I just found a new problem.

Reply:
@kacf ok, the current idea is much more simple, than what I did earlier today. main principle:
* when the `BootstrapArtifactFile` does not point to an existing file, let's only do a database initialization in `commonInit`, for the show-provides case. the rest stays exactly as it was. ready for review.